src/gui/dialogs/editor/custom_tod.cpp

```diff
@@ -168,17 +169,52 @@ void custom_tod::select_file(const std::string& default_dir)
 		dn = dlg.path();
 
 		if(data.first == "image") {
-			times_[current_tod_].image = dn;
+			times_[current_tod_].image = to_wesnoth_path(dn, "images");
 		} else if(data.first == "mask") {
-			times_[current_tod_].image_mask = dn;
+			times_[current_tod_].image_mask = to_wesnoth_path(dn, "images");
 		} else if(data.first == "sound") {
-			times_[current_tod_].sounds = dn;
+			times_[current_tod_].sounds = to_wesnoth_path(dn, "sounds");
 		}
 	}
 
 	update_selected_tod_info();
 }
 
+std::string custom_tod::to_wesnoth_path(std::string abs_path, std::string asset_type, bool copy_if_outside)
```

Comment:
This seems like a more generic function that shouldn't be in a dialog. If the name contains something about addons/editor and assets and the else case throws an exception or returns an invalid optional or something instead, I think it could go into filesystem.

In the else case it's odd that it asks the user to copy and then possibly doesn't do it anyway regardless what the user decided.

Reply:
> In the else case it's odd that it asks the user to copy and then possibly doesn't do it anyway regardless what the user decided.

What do you mean…? As far as I can tell that's only the case if the addon ID is empty, which shouldn't happen if I understand correctly.

Side note: This is minor, but it might be nice to allow the user to specify a subdirectory to copy it to. For example, if they type in "scenery/" then it will be copied to "$addondir/images/scenery/".

Reply:
If either `copy_if_outside` is false or `addon_id_` is empty the user is asked and then nothing is done.

Reply:
> This seems like a more generic function that shouldn't be in a dialog. If the name contains something about addons/editor and assets and the else case throws an exception or returns an invalid optional or something instead, I think it could go into filesystem.

Good idea. But can the gui part (the confirm dialog) go into filesystem? (Because I need it most of the time)

Reply:
> But can the gui part (the confirm dialog) go into filesystem?

I'm not really sure what you mean by this…? It's just the standard confirm dialog, right?

Reply:
> I'm not really sure what you mean by this…? It's just the standard confirm dialog, right?
Yes, it's a standard dialog, but I though it is a convention to not have gui related stuff in fliesystem.cpp. I wonder if we can make an exception for this. Perhaps @soliton- can clarify.

Reply:
I see… I'm not sure about that, then.

Reply:
> > This seems like a more generic function that shouldn't be in a dialog. If the name contains something about addons/editor and assets and the else case throws an exception or returns an invalid optional or something instead, I think it could go into filesystem.
> 
> Good idea. But can the gui part (the confirm dialog) go into filesystem? (Because I need it most of the time) Otherwise I might still need wrapper here to show the gui.

No, hence "the else case throws an exception or returns an invalid optional or something instead". The GUI part would stay in the dialog.

Reply:
Moved to `filesystem` in ee87304efde70de15eb3cb10c1416efb378b26c9.